temperature_regulator/server_thread.py

```diff
@@ -0,0 +1,55 @@
+from threading import Thread, Lock
+import socket
+import select
+from typing import Tuple
+
+class ServerThread(Thread):
+
+    RECV_MSG_MAX_SIZE = 1024
+    TEXT_ENCODING = 'UTF-8'
+    CLIENT_TIMEOUT = 3
+
+    _print_lock = Lock()
+
+    def __init__(self, id: int, connection_socket: socket.socket, client_address_pair: Tuple[str, str]):
+        Thread.__init__(self)
+        if connection_socket:
+            self._connection_socket = connection_socket
+            self._connection_socket.settimeout(ServerThread.CLIENT_TIMEOUT)
+            self._client_address, self._client_port = client_address_pair
+            self._client_port = int(self._client_port)  # Change str to int
+            self._id = id
+        else:
+            raise RuntimeError("No socket specified")
+
+    def run(self):
+        ServerThread._threaded_print(f"Thread {self._id} started running. Client address: {self._client_address} Client port: {self._client_port}")
+        try:
+            data = self._get_data_from_client()
+        except socket.timeout:
+            ServerThread._threaded_print(f"No data to read for a long time!")
+        else:
+            self._process_data(data)
```

Comment:
Czy po timeoucie ten kod się wykona (nie wiem za bardzo co robi `else` w try excepcie, piszę jakby to było takie `finnaly` javy)? Bo jeżeli tak to w sekcji timeoutu warto dodać słowo `return`

Reply:
Tak, wykona się. Else działa tak, że jego blok wykonuje się, gdy nie wyskoczy żaden wyjątek.

Reply:
W else znajduje się blok przetwarzania danych, więc moim zdaniem nie ma sensu dawać return. Chcę aby została wyświetlona wiadomość o zakończeniu działania tak jak po przetwarzaniu danych.

Reply:
ok to źle zrozumiałem co robi block `else`